### PR TITLE
Deployment Details

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Added a positional in the 'Deployments' command `<deployment-id>`.
`<deployment-id>` will get the details of the deployment, including versioned script, bindings, and usage model information.

What this PR solves / how to test:
1. Pull down PR
2. Run `wrangler deployments <insert-deployment-id> 
3. output should look something like this:

```ts
{
    Tag: '',
    Number: 0,
    'Metadata.author_id': 'Picard-Gamma-6-0-7-3',
    'Metadata.author_email': 'picard@vinyard.com',
    'Metadata.source': 'wrangler',
    'Metadata.created_on': '2021-01-01T00:00:00.000000Z',
    'Metadata.modified_on': '2021-01-01T00:00:00.000000Z',
    'resources.script': {
      etag: 'mock-e-tag',
      handlers: [ 'fetch' ],
      last_deployed_from: 'wrangler'
    },
    'resources.bindings': []
}

export default {
  async fetch(request) {
     return new Response('Hello World from Deployment 1701-E');
    },
};
```

Associated docs issues/PR:

- [ ] TODO: [insert associated docs issue(s)/PR(s)]
- [x] Tests
- [x] Changeset

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

Internal Ticket.

### CODE REVIEW

File: packages/wrangler/src/__tests__/deployments.test.ts [hunk 1]
```diff
@@ -102,4 +111,94 @@ describe("deployments", () => {
...
+
+						export default {
+							async fetch(request) {
+								return new Response('Hello World from Deployment 1701-E');
+							},
+						};"
+		`);
+		});
+	});
+	describe("rollback subcommand", () => {
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/deployments.test.ts:156]:
```suggestion

	describe("rollback subcommand", () => {
```

File: packages/wrangler/src/__tests__/deployments.test.ts [hunk 2]
```diff
@@ -102,4 +111,94 @@ describe("deployments", () => {
...
+					}
+				)
+			);
+
+			await runWrangler("deployments rollback Intrepid-Class");
+			expect(std.out).toMatchInlineSnapshot(`
+			"🚧\`wrangler deployments\` is a beta command. Please report any issues to https://github.com/cloudflare/wrangler2/issues/new/choose
+
+			Successfully rolled back to deployment ID: Intrepid-Class
+			Rollbacks metadata: undefined"
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/deployments.test.ts:200]: What does `undefined` mean in this context?

File: packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts [hunk 3]
```diff
@@ -53,17 +53,68 @@ export const mswSuccessDeployments = [
...
+export const mswSuccessDeploymentScriptMetadata = [
 	rest.get(
 		"*/accounts/:accountId/workers/services/:scriptName",
 		(_, res, ctx) => {
 			return res.once(
 				ctx.json(
 					createFetchResult({
-						default_environment: { script: { last_deployed_from: "wrangler" } },
+						default_environment: {
+							script: { last_deployed_from: "wrangler", tag: "MOCK-TAG" },
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts:64]: Does EWC always provide a `tag` property in the result?

File: packages/wrangler/src/__tests__/index.test.ts [hunk 4]
```diff
@@ -31,29 +31,29 @@ describe("wrangler", () => {
...
+			  wrangler queues                       🇶 Configure Workers Queues
+			  wrangler r2                           📦 Interact with an R2 store
+			  wrangler dispatch-namespace           📦 Interact with a dispatch namespace
+			  wrangler d1                           🗄  Interact with a D1 database
+			  wrangler pubsub                       📮 Interact and manage Pub/Sub Brokers
+			  wrangler login                        🔓 Login to Cloudflare
+			  wrangler logout                       🚪 Logout from Cloudflare
+			  wrangler whoami                       🕵️  Retrieve your user info and test your auth config
+			  wrangler types                        📝 Generate types from bindings & module rules in config
+			  wrangler deployments [deployment-id]  🚢 Displays the 10 most recent deployments for a worker
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/index.test.ts:56]: The description is no longer valid for this command.
Perhaps what we should actually do is add a `view` subcommand if you want to look at a single deployment by id?

File: packages/wrangler/src/deployments.ts [hunk 5]
```diff
@@ -91,3 +136,46 @@ function sourceStr(source: string): string {
...
+		);
+
+		logger.log(`Successfully rolled back to deployment ID: ${deploymentId}`);
+		logger.log("Rollbacks metadata:", rollbackResponse.metadata);
+	} catch (e) {
+		console.log(e);
+	}
+}
+
+export async function initializeDeployments(
```
Comment (petebacondarwin) [inline packages/wrangler/src/deployments.ts:166]: I am confused by the name of this function. What is being initialized exactly?

**Review by petebacondarwin — COMMENTED** [review]
I am not sure the structure of the `deployments` command (adding a positional) is ideal.
I updated the internal ticket with more detail.

File: packages/wrangler/src/__tests__/index.test.ts (hunk 4 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/index.test.ts:56]: I agreed before you suggested it seems.

File: packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts (hunk 3 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts:64]: As far as I have seen, in examples I have gotten in tickets. I can double check, with the API contract.

**Comment by JacobMGEvans** [comment]
> I am not sure the structure of the `deployments` command (adding a positional) is ideal. I updated the internal ticket with more detail.
@petebacondarwin Thanks for the review! 

I am superseding this PR with another that will have 2 tickets in one PR handling `deployments view <ID>` & `deployments rollback <ID>`

File: packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts (hunk 3 above)
Comment (jspspike) [reply packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts:64]: Yes

```json
{
  "result": {
    "id": "testss",
    "default_environment": {
      "environment": "wees",
      "created_on": "2022-11-18T16:40:48.50545Z",
      "modified_on": "2023-01-20T18:08:47.464024Z",
      "script": {
        "created_on": "2022-11-18T16:40:48.50545Z",
        "modified_on": "2023-01-20T18:08:47.464024Z",
        "id": "testss",
        "tag": "4918777f032148a59feb6d2c2b770e34",
        "tags": null,
        "deployment_id": "",
        "logpush": false,
        "etag": "13a3240e8fb414561b0366813b0b8f42b3e6cfa0d9e70e99835dae83d0d8a794",
        "handlers": [
          "fetch"
        ],
        "last_deployed_from": "api",
        "usage_model": "bundled"
      }
    },
    "created_on": "2022-11-18T16:40:48.50545Z",
    "modified_on": "2023-01-20T18:08:47.464024Z",
    "usage_model": "",
    "environments": [
      {
        "environment": "wees",
        "created_on": "2022-11-18T16:40:48.50545Z",
        "modified_on": "2023-01-20T18:08:47.464024Z"
      }
    ]
  },
  "success": true,
  "errors": [],
  "messages": []
}
```

**Pull request closed without merging**